runtime/nvqir/CircuitSimulator.h

```diff
@@ -514,27 +554,67 @@ class CircuitSimulatorBase : public CircuitSimulator {
     // Get a new qubit index
     auto newIdx = tracker.getNextIndex();
 
+    if (isInBatchMode()) {
+      batchModeCurrentNumQubits++;
+      // We have an allocated state, it has been set to |0>,
+      // we want to reuse it as is. If the state needs to grow, then
+      // we will ask the subtype to add more qubits.
```

Comment:
This comment seems wrong here as it is on top of the case where we don't need to allocate a new qubit.
```suggestion
      // In batch mode, we might already have an allocated state that 
      // has been set to |0..0>. We can reuse it as is, if the next qubit 
      // index is smaller than number of qubits of this allocated state.
```